components/tel/Core/Src/canload.c

```diff
@@ -0,0 +1,72 @@
+#include "canbusload.h"
+
+uint32_t can_total_bits = 0;
+static uint64_t counter_window = 0;
+static uint64_t current_bus_load = 0;
+uint32_t currentIdx = 0;
+uint32_t circularBuffer[CANLOAD_WINDOW_SIZE] = {0};
+
+uint32_t CANLOAD_calculate_message_bits(CAN_msg_t *msg)
+{
+    uint32_t bits = 0;
+    bits += CANLOAD_SOF_BITS;
+    if (msg->header.DLC == 0)
+    {
+        bits += CANLOAD_STANDARD_ID_BITS;
+    }
+    else
+    {
+        bits += CANLOAD_EXTENDED_ID_BITS;
+        bits += CANLOAD_SRR_BITS;
+        bits += CANLOAD_RESERVED_BIT_R1;
+    }
+    bits += CANLOAD_RTR_BITS;
+    bits += CANLOAD_IDE_BITS;
+    bits += CANLOAD_RESERVED_BIT_R0;
+    bits += CANLOAD_DLC_BITS;
+    bits += msg->header.DLC * 8;
+    bits += CANLOAD_CRC_BITS;
+    bits += CANLOAD_CRC_DELIMITER_BITS;
+    bits += CANLOAD_ACK_SLOT_BITS;      
+    bits += CANLOAD_ACK_DELIMITER_BITS;
+    bits += CANLOAD_EOF_BITS;
+    bits += CANLOAD_IFS_BITS;
+    return bits;
+}
+
+void update_sliding_window(uint32_t bits)
```

Comment:
Add function header. See the function header example of the Coding Style guide on the Wiki
![image](https://github.com/user-attachments/assets/35968217-249d-4492-aa7e-cf8f880741e1)